src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1445,6 +1449,56 @@ export class SavedObjectsRepository {
     };
   }
 
+  async removeReferencesTo(
+    type: string,
+    id: string,
+    options: SavedObjectsRemoveReferencesToOptions = {}
+  ): Promise<SavedObjectsRemoveReferencesToResponse> {
+    const { namespace, refresh } = options;
+    const allTypes = this._registry.getAllTypes().map((t) => t.name);
+
+    // we need to target all SO indices as all types of objects may have references to the given SO.
+    const targetIndices = this.getIndicesForTypes(allTypes);
+
+    const { body } = await this.client.updateByQuery(
+      {
+        index: targetIndices,
+        refresh,
+        body: {
+          script: {
```

Comment:
@legrego @jportner I guess this has already been discussed when implementing `deleteByNamespace`?

Reply:
I'll let @kobelb and @legrego correct me if I'm wrong, but my recollection is:

When we implemented Sharing Saved Objects Phase 1, we utilized a painless script for the `delete` operation. At that time we discussed this issue, and decided it was OK to have a dependency on scripts since `deleteByNamespace ` and `incrementCounter` already required it.

We wound up changing the `delete` behavior when implementing Sharing to All Spaces, and we removed the need for a painless script in that operation.

IIRC we did _not_ discuss returning any better error message in that case, I think because other parts of Kibana would break without scripting enabled so it seemed unnecessary. But my memory is fuzzy on this particular point.

Reply:
Our general stance has been that we're comfortable with certain functionality relying on scripting, because even though it can be disabled, it's rather rare and we haven't seen users complain about it. However, we should ensure that Kibana can still generally function without scripting being enabled, and when scripting isn't enabled we should fail in a reasonable manner.

Reply:
> so maybe we need to return a ScriptingDisabled error

Already implicitly done via the `createRepositoryEsClient` error wrapping.

> with a human readable message to improve the UX/UI.

Well, improving the UX would require that any client-side code calling (directly or indirectly) a server endpoint being able to throw a `ScriptingDisabled` error properly try/catch and identify them to show the appropriate error modal. We are very bad with server -> client error handling, but this feels like outside of the scope of the current PR, as ideally that behavior would be something globally managed (probably by core, even)

Reply:
> We are very bad with server -> client error handling

With Server --> Client communication 🙈 Complex setup that we've done in licensing is another example of such needs. Agree it deserves a separate issue. Would you mind creating one?

Reply:
I'm not sure if we should have generic client error handling for all saved objects / api's. For some domains getting a conflict error might be expected and there's no need to show a dialogue to the user.

I was thinking something like:
https://github.com/elastic/kibana/blob/b5717e0b64b1f9a5cfea3c133c1be9ac7110c075/src/plugins/visualize/public/application/utils/get_top_nav_config.tsx#L150-L163

Reply:
Created https://github.com/elastic/kibana/issues/82063